react/week2/todolist/src/App.js

```diff
@@ -0,0 +1,95 @@
+import './App.css';
+import React, { useEffect, useState } from "react";
+
+const todos = [
+    {
+        id: 1,
+        description: "Get out of bed",
+    },
+    {
+        id: 2,
+        description: "Brush teeth",
+    },
+    {
+        id: 3,
+        description: "Eat breakfast",
+    },
+];
+
+const myTodoList = ["Finish my hyf homework", "Walk the dog", "Do laundry", "Prepare for class"]
+
+function TodoItem(props) {
+    const [todoStatus, setTodoStatus] = useState(true);
+    function checkState() {
+        setTodoStatus(!todoStatus)
+    }
+
+    return (
+        <div className="todo-item">
+            <div className="item-description">
+                <input type="checkbox" onClick={checkState}></input>
+                <li style={{ 'textDecoration': todoStatus ? 'none' : 'line-through' }} >{props.description}</li>
+            </div>
+            <div>
+                <button onClick={() => props.deleteMethod(props.id)}> Delete </button>
+            </div>
+        </div>
+    )
+}
+
+function ListContainer() {
+    return (
+        <div className="list-container">
+            <h1>Todo List</h1>
+            <List />
+        </div>
+    )
+}
+
+function List() {
+    const [todo, setTodo] = useState(todos)
+    function addTodo() {
+        const randomIndex = Math.floor(Math.random() * myTodoList.length);
+        setTodo(prev => [...prev, {
+            id: prev.length + 1,
```

Comment:
Setting the id like this becomes a problem. Can you see why?

Reply:
I see, the id can be an existing one and two children can't have the same key id.